WebDriverAgentLib/Utilities/FBXPath.m

```diff
@@ -0,0 +1,259 @@
+  }
+  if (element.wdValue) {
+    id value = element.wdValue;
+    NSString *stringValue;
+    if ([value isKindOfClass:[NSValue class]]) {
+      stringValue = [value stringValue];
+    } else if ([value isKindOfClass:[NSString class]]) {
+      stringValue = value;
+    } else {
+      stringValue = [value description];
+    }
+    rc = xmlTextWriterWriteAttribute(writer, BAD_CAST "value",
+                                     [FBXPath xmlCharPtrForInput:[stringValue cStringUsingEncoding:NSUTF8StringEncoding]]);
+    if (rc < 0) {
+      [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+      return rc;
+    }
+  }
+  if (element.wdName) {
+    rc = xmlTextWriterWriteAttribute(writer, BAD_CAST "name",
+                                     [FBXPath xmlCharPtrForInput:[element.wdName cStringUsingEncoding:NSUTF8StringEncoding]]);
+    if (rc < 0) {
+      [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+      return rc;
+    }
+  }
+  if (element.wdLabel) {
+    rc = xmlTextWriterWriteAttribute(writer, BAD_CAST "label",
+                                     [FBXPath xmlCharPtrForInput:[element.wdLabel cStringUsingEncoding:NSUTF8StringEncoding]]);
+    if (rc < 0) {
+      [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+      return rc;
+    }
+  }
+  rc = xmlTextWriterWriteAttribute(writer, BAD_CAST "isVisible", element.wdVisible ? BAD_CAST "1" : BAD_CAST "0");
+  if (rc < 0) {
+    [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+    return rc;
+  }
+  rc = xmlTextWriterWriteAttribute(writer, BAD_CAST "isEnabled", element.wdEnabled ? BAD_CAST "1" : BAD_CAST "0");
+  if (rc < 0) {
+    [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+    return rc;
+  }
+  
+  rc = xmlTextWriterWriteAttribute(writer, [FBXPath xmlCharPtrForInput:[kXMLIndexPathKey cStringUsingEncoding:NSUTF8StringEncoding]],
+                                   [FBXPath xmlCharPtrForInput:[indexPath cStringUsingEncoding:NSUTF8StringEncoding]]);
+  if (rc < 0) {
+    [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterWriteAttribute. Error code: %d", rc];
+    return rc;
+  }
+  
+  return 0;
+}
+
++ (int)generateXMLPresentation:(id<FBElement>)root indexPath:(NSString *)indexPath elementStore:(NSMutableDictionary *)elementStore writer:(xmlTextWriterPtr)writer
+{
+  int rc = xmlTextWriterStartElement(writer, [FBXPath xmlCharPtrForInput:[root.wdType cStringUsingEncoding:NSUTF8StringEncoding]]);
+  if (rc < 0) {
+    [FBLogger logFmt:@"Failed to invoke libxml2>xmlTextWriterStartElement. Error code: %d", rc];
+    return rc;
+  }
+  
+  rc = [FBXPath recordElementAttributes:writer forElement:root indexPath:indexPath];
+  if (rc < 0) {
+    return rc;
+  }
+  
+  NSArray *children = @[];
+  if ([root respondsToSelector:@selector(children)]) {
```

Comment:
This is a pure hack. Instead it might make sense to extend `FBElement` protocol and add `children`, tho probably `XCUIElement` does not implement it.